lib/blacklight.rb

```diff
@@ -44,24 +47,68 @@ class << self
   # other services (e.g. refworks callback urls)
   mattr_accessor :secret_key
   @@secret_key = nil
-  
+
+  # @deprecated
   def self.solr_file
     "#{::Rails.root.to_s}/config/solr.yml"
   end
-  
+
+  def self.blacklight_config_file
+    "#{::Rails.root.to_s}/config/blacklight.yml"
+  end
+
   def self.add_routes(router, options = {})
     Blacklight::Routes.new(router, options).draw
   end
 
   def self.solr
-    @solr ||=  RSolr.connect(Blacklight.solr_config)
+    Deprecation.warn Blacklight, "Blacklight.solr is deprecated and will be removed in 6.0.0. Use Blacklight::SolrRepository#connection instead", caller
```

Comment:
@cbeer I will make this the "default" connection, but we want to avoid confusion if we have different repositories in different controllers.